### PR TITLE
update mkdocs.yml with new md filenames

### PR DESCRIPTION
**Pull request by helenlly**

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -16,39 +16,40 @@ pages:
   - Documentation: release-docs.md
 - Programming Guide: 
   - Pipeline APIs:
-    - DataFrame and ML Pipeline: programming-guide.md
-    - Autograd: programming-guide.md
-    - Transfer Learning: programming-guide.md
+    - DataFrame and ML Pipeline: ProgrammingGuide/DataFrameandMLPipeline.md
```
Comment (jason-dai) [inline docs/mkdocs.yml]: ProgrammingGuide/nnframes.md

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -16,39 +16,40 @@ pages:
...
+    - Autograd: ProgrammingGuide/autograd.md
+    - Transfer Learning: ProgrammingGuide/transferlearning.md
   - Feature Engineering:
-    - Working with Images: programming-guide.md
+    - Working with Images: ProgrammingGuide/workingwithimages.md
+    - Pre-processing: ProgrammingGuide/pre-processing.md
```
Comment (jason-dai) [inline docs/mkdocs.yml]: not needed for programming guide

File: docs/mkdocs.yml [hunk 3]
```diff
@@ -16,39 +16,40 @@ pages:
...
+    - Run: ScalaUserGuide/run.md
     - Examples: ScalaUserGuide/examples.md
 - API Guide:
   - Pipeline APIs
     - NNFrames: APIGuide/PipelineAPI/nnframes.md
-    - Autograd: APIGuide/PipelineAPI/nnframes.md
-    - Net: APIGuide/PipelineAPI/nnframes.md
+    - Autograd: APIGuide/PipelineAPI/autograd.md
+    - Net: APIGuide/PipelineAPI/net.md
   - Feature Engineering
```
Comment (jason-dai) [inline docs/mkdocs.yml]: add Preprocessing

File: docs/mkdocs.yml (hunk 1 above)
Comment (helenlly) [reply docs/mkdocs.yml]: done

File: docs/mkdocs.yml (hunk 2 above)
Comment (helenlly) [reply docs/mkdocs.yml]: removed

File: docs/mkdocs.yml (hunk 3 above)
Comment (helenlly) [reply docs/mkdocs.yml]: added

**Pull request merged**